### DISCUSSION TITLE
How to apply relation condition on the last record retrieved from that relation.

### DISCUSSION THREAD

**Discussion by kevinmamaqi** [prisma/prisma#13556] [Q&A] [answered]
I have the following query, from which I want to get the last record with orderBy: createdAt and take:1 

```
const query = {
  where: {
      {
        log_table: {
          some: {  <----- Apply here to only the last row
            info: { 
              path: ["date"],
              gt: d.start.endOf("day").valueOf(),
              lte: d.end.endOf("day").valueOf(),
            },
          },
        },
      },
  },
  include: {
    log_table: {
      take: 1,
      orderBy: {
        createdAt: "desc",
      },
    },
  },
};

```

**Comment by nurul3101** [comment, marked as answer]
Hey @kevinmamaqi 👋 
You could get the last record passing -1 in the `take` attribute, passing a negative number reverses the list.
Here's a reference for take parameter: [reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#remarks-4)

**Comment by ricardo-valero** [comment]
Hi @kevinmamaqi, how did you solve this?
Using `{ take: -1 }` instead of `{ take: 1, orderBy: { createdAt: "desc" } }` is fine
but your question about if the `where` clause applies **only** to the records in the `include`...
I think it doesn't work this way
I'm curious because I need something like this

A quick dumb example

This doesn't work as I'd expect
```ts
     prisma.company.findMany({
        where: { archives: { every: { value: false } } },
        include: { archives: { take: -1 } },
      }),
```

This does but by filtering with javascript 
```ts
      prisma.company.findMany({
        include: { archives: { take: -1 } },
      }).then(
        companies => companies.filter(
          company => company.archives.length === 0
          || company.archives.every(archive => archive.value === false)),
      ),
```
